Assets/Tests/PlayModeTests/DeviceIdTests.cs

```diff
@@ -377,6 +372,76 @@ public void GeneratedDeviceID_CustomDeviceIDProvided()
             ValidateDeviceIDAndType(Countly.Instance, deviceID, DeviceIdType.SDKGenerated);
         }
 
+        // 'SetId' method in DeviceIdCountlyService
+        // We provide empty, null, whitespace and valid ids to SetId method over DeveloperProvided id
+        // Id should only change in case of a valid id
+        [TestCase("", DeviceIdType.DeveloperProvided, false)]
+        [TestCase(null, DeviceIdType.DeveloperProvided, false)]
+        [TestCase(" ", DeviceIdType.DeveloperProvided, false)]
+        [TestCase("new_user", DeviceIdType.DeveloperProvided, true)]
+        public void SetId_OverDeveloperProvided(string id, DeviceIdType expectedIdType, bool isExpectingChange)
+        {
+            Countly.Instance.Init(TestUtility.CreateBaseConfig());
+            string deviceID = Countly.Instance.Device.DeviceId; // "test_user"
+
+            ValidateDeviceIDAndType(Countly.Instance, deviceID, DeviceIdType.DeveloperProvided, true);
+            Countly.Instance.Device.SetId(id);
+            if (isExpectingChange) {
+                ValidateDeviceIDAndType(Countly.Instance, id, expectedIdType, true);
+            } else {
+                ValidateDeviceIDAndType(Countly.Instance, deviceID, expectedIdType, true);
+            }
+        }
+
+        // 'SetId' method in DeviceIdCountlyService
+        // We provide empty, null, whitespace and valid ids to SetId method over SDKGenerated id
+        // Id should only change in case of a valid id
+        [TestCase("", DeviceIdType.SDKGenerated, false)]
+        [TestCase(null, DeviceIdType.SDKGenerated, false)]
+        [TestCase(" ", DeviceIdType.SDKGenerated, false)]
+        [TestCase("new_user", DeviceIdType.DeveloperProvided, true)]
+        public void SetId_OverSDKGenerated(string id, DeviceIdType expectedIdType, bool isExpectingChange)
+        {
+            CountlyConfiguration config = new CountlyConfiguration(_appKey, _serverUrl);
+            Countly.Instance.Init(config);
```

Comment:
Why aren't we just using TestUtility.CreateBaseConfig() instead of making a whole new config?